src/overlays/actors/ovl_Obj_Bean/z_obj_bean.c

```diff
@@ -27,160 +72,890 @@ const ActorInit Obj_Bean_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_8093868C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938704.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938728.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_8093876C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938780.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938804.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938834.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938874.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_809388A8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_8093892C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938958.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938998.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_809389BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938A14.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938A5C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938AA4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938AD8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938C1C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/ObjBean_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938E00.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Bean/func_80938F50.s")
+void ObjBean_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjBean* this = THIS;
+    s32 sp2C = OBJBEAN_GET_C000(&this->dyna.actor);
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->unk_1FE = 0;
+    this->unk_1B8 = 0.1f;
+    DynaPolyActor_Init(&this->dyna, 3);
+    Collider_InitCylinder(globalCtx, &this->collider);
+
+    if ((sp2C == ENOBJBEAN_GET_C000_1) || (sp2C == ENOBJBEAN_GET_C000_2)) {
+        this->dyna.actor.update = func_80938C1C;
+        this->dyna.actor.textId = 0xFD;
+        if (sp2C == ENOBJBEAN_GET_C000_1) {
+            Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit2);
+            Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+        }
+        this->dyna.actor.flags |= 0x10;
+        func_80937C10(this);
+        if (!func_80936D58(this, globalCtx)) {
+            Actor_MarkForDeath(&this->dyna.actor);
+        } else {
+            func_800BC154(globalCtx, &globalCtx->actorCtx, &this->dyna.actor, 7);
+            func_80937DD8(this);
+        }
+    } else {
+        s32 params2 = OBJBEAN_GET_3F00(&this->dyna.actor);
+        Path* path = &globalCtx->setupPathList[params2];
+
+        this->unk_1DE = this->dyna.actor.home.rot.z & 3;
```

Comment:
I know you asked about this in the discord. Personally since it is acting like a params access, I think we should macro these.